### DISCUSSION TITLE
Realtime with custom headers

### DISCUSSION THREAD

**Discussion by ksamirdev** [pocketbase/pocketbase#3571] [Q&A]
How would I subscribe to a record with custom headers?
Actually, I'm using pocketbase for the `realtime` feature. 
I have `@collection.process.userid = @request.headers.userid` set to `View Rule`

**Comment by ganigeorgiev** [comment]
It is not possible at the moment. It is planned for the next release together with the `filter` and `expand` request parameters support.

_Side-note: I don't know what the above rule us supposed to do, but note that the `@collection.*` rule is intended to target collection fields that are not directly related to your current one (aka. to act as a join). If with `@collection.process.userid` you are targetting the current collection, then you can replace it with just the field name, aka. `userid = @request.headers.userid`._

**Reply by ksamirdev** [reply]
I changed API rule. 
Btw I tried creating my custom method but it says "Something went wrong while processing your request.". It is successfully connected to SSE

```ts
const eventSource = new EventSource("http://127.0.0.1:8090/api/realtime");

eventSource.onopen = (ev) => {
  console.log("CONNECTED!");
};

eventSource.addEventListener("PB_CONNECT", async (e) => {
  const data = JSON.parse(e.data) as unknown as { clientId: string };
  await fetch("http://127.0.0.1:8090/api/realtime", {
    method: "POST",
    body: JSON.stringify({
      clientId: data.clientId,
      subscriptions: ["2ndni9cgxqb5bla"], // record id
    }),
    headers: {
      userid: "hdvjqu7l3inbki0",
    },
  });
});
```

**Reply by ganigeorgiev** [reply]
Sending headers (and query parameters) with the realtime topic(s) is not supported on the server-side, it is not only a frontend limitation.

**Reply by ksamirdev** [reply]
Understood! Any ETA when it will be available?

**Reply by ganigeorgiev** [reply]
I'm planning this weekend to start experimenting with couple options but there no fixed ETAs and I don't want to make any promises.

**Reply by ganigeorgiev** [reply]
_@samocodes Just to let you know that I've pushed in the `develop` branch experimental `filter`, `expand`, `fields`, custom query and headers parameters support for the realtime subscription. There are some minor polishing details that still needs to be done, but you can test it with the [PocketBase v0.20.0-rc prerelease](https://github.com/pocketbase/pocketbase/releases/tag/v0.20.0-rc) (requires also JS SKD v0.20.0-rc or Dart SDK v0.17.0-rc)._
